Tests/test_file_tiff.py

```diff
@@ -450,6 +450,18 @@ def test_save_tiff_with_jpegtables(self):
         # Should not raise UnicodeDecodeError or anything else
         im.save(outfile)
 
+    def test_open_tiff_uint16_multiband(self):
+        """Test opening multiband TIFFs and reading all channels."""
+        base_value = 4660
+        for i in range(2, 6):
+            infile = "Tests/images/uint16_{}_{}.tif".format(i, base_value)
+            im = Image.open(infile)
+            im.load()
+            for j in range(i):
+                current_value = base_value + j
+                im.seek(j)
+                self.assertEqual(im.getpixel((0, 0)), current_value)
```

Comment:
I don't think that this test is doing what you think it's doing. 

The seek command is paging through the tiff, as if going from frame to frame in an animation, and looking for a single integer from the pixel location. That's not the kind of tiff you're providing. (and it's currently supported, if that's what you have for actual images)

The tiffs in the test are stored as 2,3,4,5 channel tiffs, where each pixel is an n tuple. 

Furthermore, the last channel in each case is 0, so for the 2 channel image, the pixel would be (4660, 0), and the 5 would be (4660, 4661, 4662, 4663, 0).